scripts/generator/graphql_generator/csharp/type.cs.erb

```diff
@@ -146,6 +146,27 @@ namespace <%= namespace %>.GraphQL {
                     }
                 <% end %>
             }
+
+            <%= params_doc(type.input_fields) %>
+            public <%= type.classify_name %>(Dictionary<string, object> dataJSON) {
+                <%# handle adding required fields %>
+                <% if type.required_input_fields.length > 0 %>
+                try {
+                    <% type.required_input_fields.each do |field| %>
+                        Set("<%= field.name %>", dataJSON["<%= escape_reserved_word(field.name) %>"]);
+                    <% end %>
+                } catch {
+                    throw;
```

Comment:
Could be more nicely phrased though, with perhaps some basic info on how to fix.